src/OrchardCore.Modules/OrchardCore.Forms/Shapes.cs

```diff
@@ -0,0 +1,123 @@
+            {
+
+                dynamic cardShape = context.Shape;
+
+                IShape settingTool = await context.ShapeFactory.New.ToolFormSettings();
+                settingTool.Metadata.Name = "ToolFormSettings";
+
+                settingTool.Properties["Content"] = await context.ShapeFactory.CreateAsync("ContentZone");
+                settingTool.Properties["Actions"] = await context.ShapeFactory.CreateAsync("ContentZone");
+
+                cardShape.Footer.Add(settingTool, "before");
+
+
+                IShape formWorkflowTool = await context.ShapeFactory.New.ToolFormWorkflow();
+                formWorkflowTool.Metadata.Name = "ToolFormWorkflow";
+
+                formWorkflowTool.Properties["Content"] = await context.ShapeFactory.CreateAsync("ContentZone");
+                formWorkflowTool.Properties["Actions"] = await context.ShapeFactory.CreateAsync("ContentZone");
+
+
+                cardShape.Footer.Add(formWorkflowTool, "before");
+
+            })
+            .OnDisplaying(context =>
+            {
+                dynamic cardShape = context.Shape;
+
+                bool hasSettingsTool = false;
+
+                if (cardShape.ContentItem.ContentType == "Form")
+                {
+                    //Move WorkflowSettings to ToolFormWorkflow Content, if any
+                    foreach (var item in cardShape.ContentEditor.WorkflowSettings)
+                    {
+                        cardShape.Footer.ToolFormWorkflow.Content.Add(item);
+                    }
+                    cardShape.ContentEditor.WorkflowSettings = null;
+
+                    //Configure ToolFormWorkflow Shape
+                    cardShape.Footer.ToolFormWorkflow.ModalId = $"ToolFormWorkflow_{cardShape.ContentItem.ContentItemId}";
+                    cardShape.Footer.ToolFormWorkflow.ContentItemDisplayText = cardShape.ContentItem.DisplayText;
+                    cardShape.Footer.ToolFormWorkflow.ContentType = cardShape.ContentItem.ContentType;
+
+                    hasSettingsTool = true;
+                }
+                else
+                {
+                    // Remove Workflow Tool
+                    cardShape.Footer.Remove("ToolFormWorkflow");
+
+                    if (cardShape.ContentItem.ContentType == "ValidationSummary")
+                    {
+
+                        cardShape.IsCollapsible = false;
+
+                        // Remove Toggle All and Settings for Validation Summary
+                        cardShape.Footer.Remove("ToolToggleWidgets");
+
+                    }
+
+                    else if (cardShape.ContentItem.ContentType == "Label" ||
+                                cardShape.ContentItem.ContentType == "Button" ||
+                                cardShape.ContentItem.ContentType == "TextArea" ||
+                                cardShape.ContentItem.ContentType == "Input" ||
+                                cardShape.ContentItem.ContentType == "Validation"
```

Comment:
Or doing this. What happens when someone adds a custom form content type - everything breaks

Reply:
I don't think it will break. It will render default `Content_Edit` 

IMO this is in scope of module to customize editing experience . i.e this code is only related to `Forms` module and editing experience is defined for `ContentType `defined by `Forms` Module only.

If someone creates new module and create custom content type using Parts from `Form`, they'll have to define their custom editing experience in their module if they don't like default `Content_Edit`